examples/quad/main.rs

```diff
@@ -359,9 +360,18 @@ fn main() {
             .unwrap_or(formats[0])
     });
 
-    let swap_config = SwapchainConfig::from_caps(&caps, format);
+    let (min_width, max_width) = (caps.extents.start.width, caps.extents.end.width - 1);
+    let (min_height, max_height) = (caps.extents.start.height, caps.extents.end.height - 1);
+
+    let mut swap_config = SwapchainConfig::from_caps(&caps, format);
```

Comment:
maybe we need to add a parameter to `from_caps` specifying the default sizes, so that the branch can be moved into the implementation?

Reply:
I added a `default_extent` param to `SwapchainConfig::from_caps`. If the passed in parameter doesn't fit within the range of the surface sizes, than it's clamped to fit automatically. I also added a debug assertion if it was clamped automatically and the size was changed to fit.